test/index.spec.ts

```diff
@@ -137,452 +137,540 @@ test('encryptor:encrypt & decrypt with wrong password', async ({ page }) => {
-    },
-  );
+      // Prove that a vault created with key can be decrypted with password
+      const decryptedResult = await page.evaluate(
+        async (args) =>
+          await window.encryptor.decryptWithDetail(args.password, args.data),
+        { password, data: JSON.stringify(decryptable) },
+      );
 
-  expect(JSON.stringify(decryptedResult)).toStrictEqual(
-    JSON.stringify(newData),
-  );
-});
+      expect(JSON.stringify(decryptedResult.vault)).toStrictEqual(
+        JSON.stringify(newData),
+      );
+    });
+
+    test('encryptor:encryptWithKey works with decryptWithKey', async ({
+      page,
+    }) => {
+      const password = 'a sample passw0rd';
+      const startingData = { foo: 'data to encrypt' };
+
+      // Get an exported key to use
+      const exportedKeyString = await page.evaluate(
+        async (args) => {
+          const { exportedKeyString: newKeyString } =
+            await window.encryptor.encryptWithDetail(args.password, args.data);
+
+          return newKeyString;
+        },
+        { data: startingData, password },
+      );
 
-test('encryptor:keyFromPassword cannot be exported by default', async ({
-  page,
-}) => {
-  const password = 'a sample passw0rd';
-  const data = { foo: 'data to encrypt' };
-  const salt = await page.evaluate(() => window.encryptor.generateSalt());
+      // Update the data, encrypt using key
+      const newData = { ...startingData, bar: 'more data' };
+      const encryptWithKeyResult = await page.evaluate(
+        async (args) => {
+          const key = await window.encryptor.importKey(args.keyString);
+          const result = await window.encryptor.encryptWithKey(key, args.data);
+
+          return {
+            encryptWithKeyResult: result,
+            exportedKeyString: await window.encryptor.exportKey(key),
+          };
+        },
+        { data: newData, keyString: exportedKeyString },
+      );
 
-  const exportResult = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.keyFromPassword(
-        args.password,
-        args.salt,
+      // Prove that a vault created with key can be decrypted with password
+      const decryptedResult = await page.evaluate(
+        async (args) => {
+          const key = await window.encryptor.importKey(args.exportedKeyString);
+          return await window.encryptor.decryptWithKey(key, args.data);
+        },
+        {
+          exportedKeyString: encryptWithKeyResult.exportedKeyString,
+          data: encryptWithKeyResult.encryptWithKeyResult,
+        },
       );
 
-      try {
-        const result = await window.encryptor.exportKey(key);
-        return result;
-      } catch (e) {
-        return 'error';
-      }
-    },
-    { data, password, salt },
-  );
+      expect(JSON.stringify(decryptedResult)).toStrictEqual(
+        JSON.stringify(newData),
+      );
+    });
+
+    test('encryptor:keyFromPassword cannot be exported by default', async ({
```

Comment:
This test doesn't rely on either `sampleEncryptedData` or `oldSampleEncryptedData`. Should it be inside the loop?